lang/LangSource/PyrObject.cpp

```diff
@@ -1211,33 +1222,53 @@ void buildBigMethodMatrix()
 		}
 		return classes;
 	};
-
+#ifdef _MSC_VER
+	auto filledClassIndices = std::async( std::launch::async, fillClassIndices, classes);
+#else
 	auto filledClassIndices = boost::async( pool, fillClassIndices, classes);
+#endif
 
 	bigTable = (PyrMethod**)pyr_pool_compile->Alloc(bigTableSize * sizeof(PyrMethod*));
 	MEMFAIL(bigTable);
 
+#ifndef _MSC_VER
 	pool.try_executing_one();
+#endif	
 	filledClassIndices.wait();
+#ifdef _MSC_VER
+	size_t numentries = fillClassRows(class_object, bigTable);
+#else
 	size_t numentries = fillClassRows(class_object, bigTable, pool);
+#endif
 	post("\tnumentries = %lu / %d = %.2g\n", numentries, bigTableSize, (double)numentries/(double)bigTableSize);
 
 
 	ColumnDescriptor * filledSelectors = filledSelectorsFuture.get();
-
+#ifdef _MSC_VER
+	std::vector< std::future<void> > columnDescriptorsWithStats;
+#else
 	std::vector< boost::future<void> > columnDescriptorsWithStats;
+#endif
 	size_t selectorsPerJob = numSelectors / cpuCount / 2;
 	for( size_t beginSelectorIndex : boost::irange(selectorsPerJob, numSelectors, selectorsPerJob) ) {
 		size_t endSelectorIndex = std::min( beginSelectorIndex + selectorsPerJob, numSelectors );
+#ifdef _MSC_VER
+		auto future = std::async( std::launch::async, calcRowStats, bigTable, filledSelectors, numClasses, numSelectors, beginSelectorIndex, endSelectorIndex );
+#else
 		auto future = boost::async( pool, calcRowStats, bigTable, filledSelectors, numClasses, numSelectors, beginSelectorIndex, endSelectorIndex );
+#endif
 		columnDescriptorsWithStats.push_back( std::move(future) );
 	}
 
 	calcRowStats(bigTable, filledSelectors, numClasses, numSelectors, 0, std::min( selectorsPerJob, numSelectors) );
 
 	for( auto & future : columnDescriptorsWithStats ) {
+#ifdef _MSC_VER
+        future.wait();
```

Comment:
Ok, I will change the async to deferred.

Sorry about the space, I will fix that too, of course.

> >  		columnDescriptorsWithStats.push_back( std::move(future) );
> >  	}
> >
> >  	calcRowStats(bigTable, filledSelectors, numClasses, numSelectors, 0, std::min( selectorsPerJob, numSelectors) );
> >
> >  	for( auto & future : columnDescriptorsWithStats ) {
> > +#ifdef _MSC_
>
>
>